OpenRA.Mods.Common/Activities/FindAndDeliverResources.cs

```diff
@@ -128,13 +113,13 @@ public override bool Tick(Actor self)
 			// of the refinery entrance.
 			if (LastSearchFailed)
 			{
-				var lastproc = harv.LastLinkedProc ?? harv.LinkedProc;
-				if (lastproc != null && !lastproc.Disposed)
+				var lastproc = harv.DockClientManager.LastReservedHost;
+				if (lastproc != null)
 				{
-					var deliveryLoc = lastproc.Location + lastproc.Trait<IAcceptResources>().DeliveryOffset;
-					if (self.Location == deliveryLoc && harv.IsEmpty)
+					var deliveryLoc = lastproc.DockPosition;
+					if (self.CenterPosition == deliveryLoc && harv.IsEmpty)
```

Comment:
Could you make this use a `DockingPosition` var that includes an optional offset instead of using the unit's `CenterPosition` directly? I am thinking of a dock where different harvester units will want different offsets so that they fit properly inside the refinery. E.g. a very large harvester might want a -20 X and Y offset to their DockingPosition, while a smaller harvester won't need any offset.

Reply:
This code is for moving the harvester away from the dock incase it idles there.

Also, Mobile operate in CPos, so you can't get any offsets regardless. If you want to offset in WPos you'll need to use dragging

Reply:
Dragging from what I understand occurs on all harvesters though doesn't it? (It is the way the unit slowly enters into the refinery) Since I can see `StartDrag` has no offset. I see you use `harvPos = centerPosition` in other parts of the code, could you make this a method `HarvPos() => self.CenterPosition`? That way if someone wanted to subclass the docking and overload `HarvPos()` with a different type of position they could. It also means they could apply offsets and the like much easier without editing multiple parts of the code base.

Reply:
I made this part convert to CPos since `FindAndDeliverResources` operates in CPos anyway

Reply:
Ok that's fine, I think decoupling it from CPos would be beyond the scope of this PR.

Reply:
Yeah,
 I also did try abstracting IsAtDock and it just felt like I couldn't extract it in a non-performance expensive way